core/src/main/java/org/apache/calcite/sql2rel/SqlToRelConverter.java

```diff
@@ -3305,39 +3385,70 @@ private Pair<RexNode, RelNode> convertOnCondition(
 
     bb.setRoot(ImmutableList.of(leftRel, rightRel));
 
-    final LookupContext old_right_rels = new LookupContext(
-        bb, ImmutableList.of(rightRel), bb.systemFieldList.size());
+    final int num_left_rels = new LookupContext(
+        bb, ImmutableList.of(leftRel), bb.systemFieldList.size()).relOffsetList.size();
 
-    replaceSubQueries(bb, condition, RelOptUtil.Logic.UNKNOWN_AS_FALSE);
-    boolean newRight = bb.root == null || bb.registered.size() == 0;
-    final RelNode newRightRel = newRight
-        ? rightRel
-        : bb.reRegister(rightRel);
 
     /**
-     * Logic needed to update the offset list, which tracks at what offsets in the
-     * relnode list we insert sub queries
-     * can be found in the flattened rel list. (see the variable for more information)
+     * Below is the code needed to handle replacing sub querries in the "ON" clause.
```

Comment:
Is this comment still relevant or should be it converted?

Reply:
It's correct, but the phrasing is weird, I updated it.